client/data/followers/use-remove-follower-mutation.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+import { useMutation, useQueryClient } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import wp from 'calypso/lib/wp';
+
+function useRemoveFollowerMutation() {
+	const queryClient = useQueryClient();
+	const { mutate, ...mutationObj } = useMutation(
+		( { siteId, type, followerId } ) => {
+			const method = type === 'email' ? 'removeEmailFollower' : 'removeFollower';
+			return wp.undocumented().site( siteId )[ method ]( followerId );
+		},
+		{
+			onSuccess( data, variables ) {
+				const { siteId, type } = variables;
+				queryClient.invalidateQueries( [ 'followers', siteId, type ] );
+			},
+		}
+	);
+
+	const removeFollower = useCallback(
+		( siteId, type, followerId ) =>
+			mutate( {
+				siteId,
+				type,
+				followerId,
+			} ),
+		[ mutate ]
+	);
+
+	return { removeFollower, mutate, ...mutationObj };
```

Comment:
Removing the `mutate` prop by `...mutationObj` destructuring and then adding it again ... you didn't need to do that. I'd personally write it as:
```js
const mutation = useMutation();
const removeFollower = useCallback( () => mutation.mutate(), [ mutation.mutate ] );
return { ...mutation, removeFollower };
```

Reply:
That doesn't seem to work properly. `useCallback` complains about missing dependencies when using it that way.

<img width="1490" alt="Screenshot 2021-03-16 at 09 08 12" src="https://user-images.githubusercontent.com/9202899/111276494-ac7b0a00-8637-11eb-89b5-4e1594e7ecde.png">

Reply:
I see, that's a deficiency of the ESLint rule -- at runtime it would work reliably.

In that case I would do:
```js
const mutation = useMutation();
const { mutate } = mutation;
...
```
But this all is just my personal style preference. I'd try to preserve the `mutation` object and only add new fields to it. Don't feel obliged to follow it 🙂

Reply:
I think that's a great suggestion 👍🏻